synapse/federation/transaction_queue.py

```diff
@@ -69,13 +70,20 @@ def __init__(self, hs, transport_layer):
         # destination -> list of tuple(edu, deferred)
         self.pending_edus_by_dest = edus = {}
 
+        self.pending_presence_by_dest = presence = {}
+        self.pending_edus_keyed_by_dest = edus_keyed = {}
```

Comment:
Why do we need separate both pending keyed edus and pending presence? Why can't presence be a keyed EDU?